libero/src/main/groovy/com/connexta/libero/Libero.groovy

```diff
@@ -56,49 +82,93 @@ class Libero {
+     * @param config
+     */
+    private void initializeConfig(Config config) {
+        config.projectName = config.projectName ?: new File(config.projectDir).getName()
+        config.sourceRemote = config.sourceRemote ?: 'origin'
+        config.destBranch = config.destRemote ?: config.sourceRemote
+        // might need to default to something like "refs/remotes/${config.sourceRemote}/master"
+        config.ref = config.ref ?: 'master'
+        config.destBranch = config.destBranch ?: 'master'
+        config.preProps = config.preProps ?: null as Map
+        config.postProps = config.postProps ?: null as Map
+        config.gitPush = config.gitPush ?: false
+        config.quickBuild = config.quickBuild ?: false
+        config.dryRun = config.dryRun ?: false
+        config.force = config.force ?: false
+    }
+
+    /**
+     * performs initial preparation of git repository
+     * @param config
+     * @param git
+     */
+    private void initialRepoPreparation(Config config, Grgit git) {
+        git.fetch(remote: config.sourceRemote)
+        git.checkout(branch: config.ref)
+        git.pull(remote: config.sourceRemote)
+    }
+
+    /**
+     * Checks config file for missing values that require computed defaults
+     * @param config
+     */
+    private void finalizeConfig(Config config) {
+        config.startVersion = util.getPomVersion(config.projectDir)
+        config.releaseVersion = config.releaseVersion ?: util.createReleaseVersion(config.startVersion)
+        config.nextVersion = config.nextVersion ?: util.incrementVersion(config.releaseVersion)
+        config.releaseName = config.releaseName ?: "${config.projectName}-${config.releaseVersion}"
     }
 
     /**
      * Executes the build and deploy of the release
      * @param config Libero configuration
      */
     def runBuild(Config config) {
-        String gitCommand = "git"
-        String gitCheckoutCommand = "${gitCommand} checkout ${config.releaseName}"
-        String gitPushCommand = "${gitCommand} push ${config.destRemote}"
-        String gitMasterCommand = "${gitCommand} checkout master"
-        String mavenCommand = "mvn -f ${config.projectDir + "/pom.xml"}"
-        String quickBuildParams = "-Dfindbugs.skip=true -DskipTests=true -Dpmd.skip=true -Djacoco.skip=true -DskipTestScope=true -DskipProvidedScope=true -DskipRuntimeScope=true"
-        String deployCommand = "${mavenCommand} ${quickBuildParams} -T2C deploy"
-        String buildCommand
-
-        // Check out the git tag
-        util.executeCommand(gitCheckoutCommand, config.projectDir)
 
+        Properties quickBuildProps = [
+                "findbugs.skip": "true",
+                "pmd.skip": "true",
+                "jacoco.skip": "true",
+                skipTestScope: "true",
+                skipProvidedScope: "true",
+                skipRuntimeScope: "true",
+                skipTests: "true"
+        ]
+        InvocationRequest buildRequest = new DefaultInvocationRequest(
```

Comment:
not sure, because `pomfile` which is an input to this `InvocationRequest` would't be known until `run()` was called